frontend/src/metabase/dashboard/constants.js

```diff
@@ -6,3 +8,7 @@ export const SIDEBAR_NAME = {
   sharing: "sharing",
   info: "info",
 };
+
+// most browsers don't use a custom message with beforeunload anymore, just putting here to retain compatibility
+// https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event#compatibility_notes
+export const BEFORE_UNLOAD_UNSAVED_MESSAGE = t`You have unsaved changes.`;
```

Comment:
Maybe we should add a hook to `metabase/core/hooks` that wraps `useBeforeUnload` with this message?